modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb

```diff
@@ -89,8 +89,22 @@ def get_attachment_ids_and_form(parsed_form_data)
       end
 
       def supporting_document_ids(parsed_form_data)
-        parsed_form_data['supporting_docs']&.pluck('attachment_id')&.compact.presence ||
-          parsed_form_data['supporting_docs']&.pluck('claim_id')&.compact.presence || []
+        cached_uploads = []
+        parsed_form_data['supporting_docs']&.each do |d|
+          # Get the database record that corresponds to this file upload:
+          record = PersistentAttachments::MilitaryRecords.find_by(guid: d['confirmation_code'])
+          # Push to our array with some extra information so we can sort by date uploaded:
+          cached_uploads.push({ attachment_id: d['attachment_id'],
+                                created_at: record.created_at,
+                                file_name: record.file.id })
+        end
+
+        # Sort by date created so we have the file's upload order and
+        # reduce down to just the attachment id strings:
+        attachment_ids = cached_uploads.sort_by { |h| h[:created_at] }.pluck(:attachment_id)&.compact.presence
+
+        # Return either the attachment IDs or `claim_id`s (in the case of form 10-7959a):
+        attachment_ids || parsed_form_data['supporting_docs']&.pluck('claim_id')&.compact.presence || []
```

Comment:
The order of the attachment IDs has always mattered, so this is ensuring we have things in a sane order (date created) rather than the previous arbitrary order.

In future we will probably want to look into a wider refactor. A direct mapping of files to doc types rather than relying on the order of this array would be better, but this is a necessary first step to make sure the MVP works.